src/main/java/org/fcrepo/upgrade/utils/BackupFixer.java

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.upgrade.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.zip.GZIPInputStream;
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.zip.GZIPOutputStream;
+import java.io.FileOutputStream;
+import java.util.HashSet;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+/**
+ * This utility is designed to be used to help migrate from 4.4, 4.5, and 4.6 to Fedora 4.7.0
+ *
+ * It should be run on the directory produced by a /fcr:backup request, prior to requesting /fcr:restore
+ *
+ * Warning: This utility is destructive (but likely not in a bad way), and should be run against a copy of the backup
+ * if the backup is irreplaceable.
+ *
+ * @author escowles
+ * @since 2016-10-28
+ */
+public class BackupFixer {
+
+    /**
+     * Private constructor
+     */
+    private BackupFixer() {
+        // prevent public construction
+    }
+
+    /**
+     * main
+     * @param args from command line
+     * @throws IOException
+     */
+    public static void main( final String[] args ) throws IOException {
+        if ( args.length < 1 ) {
+            System.err.println("usage: BackupFixer [backup directory]");
+            return;
+        }
+
+        final File dir = new File(args[0]);
+
+        if ( !dir.exists() || dir.listFiles().length == 0 ) {
+            System.err.println("Directory " + args[0] + " is empty or doesn't exist");
+            return;
+        }
+
+        final File[] files = dir.listFiles();
+        for ( int i = 0; i < files.length; i++ ) {
+            final String fn = files[i].getName();
+            if ( files[i].isFile() && fn.startsWith("documents_") && fn.endsWith(".gz") ) {
+                fix(files[i]);
+            }
+        }
+    }
+    private static void fix(final File f) throws IOException {
+        System.out.println("Fixing: " + f.getName());
+
+        // move original file
+        final File moved = new File(f.getParent(), f.getName() + ".orig");
+        if ( moved.exists() ) {
+            System.err.println("Backup file already exists, skipping: " + moved.getName());
+            return;
+        }
+        f.renameTo(moved);
+
+        // filter duplicate ids
+        PrintWriter out = null;
+        try {
+            final BufferedReader buf = new BufferedReader(new InputStreamReader(new GZIPInputStream(
+                    new FileInputStream(moved))));
+            out = new PrintWriter(new GZIPOutputStream(new FileOutputStream(f)));
+
+            final HashSet<String> ids = new HashSet<>();
```

Comment:
I have pushed a new commit with the suggested change: https://github.com/fcrepo4-exts/fcrepo4-upgrade-utils/pull/12/commits/18e93cfe4e38b0883c826c153337447b0f90f701